src/compiler/emitter.ts

```diff
@@ -4766,9 +4766,14 @@ const _super = (function (geti, seti) {
                 emitToken(SyntaxKind.CloseBraceToken, body.statements.end);
             }
 
-            function findInitialSuperCall(ctor: ConstructorDeclaration): ExpressionStatement {
+            /**
+             * Return the statement at a given index if it is a super-call statement
+             * @param ctor constructor declaration
+             * @param index an index to constructor's body to check
+             */
+            function getSuperCallAtGivenIndex(ctor: ConstructorDeclaration, index: number): ExpressionStatement {
                 if (ctor.body) {

```

Comment:
If you invert the conditions, this becomes a lot cleaner.

`if (!statements || index >= statements.length) {
    return false;
}

const statement = statements[index];
if (statement.kind !== SyntaxKind.ExpressionStatement) {
    return false;
}

return isSuperCallExpression((statement as ExpressionStatement).expression);